lib/internal/fs/recursive_watch.js

```diff
@@ -159,10 +160,11 @@ class FSWatcher extends EventEmitter {
           if (file.isFile()) {
             this.#watchFile(f);
           } else if (file.isDirectory() && !file.isSymbolicLink()) {
-            await this.#watchFolder(f);
```

Comment:
Moving to `Promise.all`, this code will not probably throw [EMFILE, too many open files](https://stackoverflow.com/questions/8965606/node-and-error-emfile-too-many-open-files)?